### PR TITLE
Fix null variables and default values

### LINKED ISSUE #239: Fix ambiguity with null variable values and default values
We need to ensure that null variable values and default values are handled correctly. The reference implementation can be found here: https://github.com/graphql/graphql-js/pull/1274/files

### PR DESCRIPTION
**Pull request by crisu83**
Fixes ambiguity with null variable values and default values (fixes #239).

### CODE REVIEW

**Comment by Jalle19** [comment]
Can you fix the tests?

**Comment by crisu83** [comment]
Note: I don't think that the null value resolution can be completely fixed so that it works exactly like in the reference implementation because it uses `undefined` for the value when it's not defined and `null` when a value is implicitly set to `null`. It might be possible to implement a fix for this, but I don't see how. @Jalle19 maybe you could take a look and give your 50 cents? Thanks!

**Comment by crisu83** [comment]
@Jalle19 the tests are passing now.

**Comment by crisu83** [comment]
I thought about the `null`/`undefined` problem overnight and I think that we could fix it by introducing a `ConversionResult` class that allows us to represent a value corresponding `undefined` in the JS implementation. Thoughts @Jalle19?

**Comment by crisu83** [comment]
I found and fixed the null conversion issue, the issue was in the test case and not in the library. Duh! 😐

File: src/Execution/ValuesResolver.php [hunk 1]
```diff
@@ -26,13 +27,22 @@
...
+use function array_key_exists;
+use function array_keys;
+use function array_map;
+use function array_merge;
+use function Digia\GraphQL\Test\jsonEncode;
 use function Digia\GraphQL\Util\find;
+use function Digia\GraphQL\Util\invariant;
 use function Digia\GraphQL\Util\keyMap;
 use function Digia\GraphQL\Util\suggestionList;
+use function implode;
```
Comment (Jalle19) [inline src/Execution/ValuesResolver.php]: I would just use \ for these core methods

Comment (crisu83) [reply src/Execution/ValuesResolver.php]: My bad. I let PhpStorm resolve ”issues” on it’s own. 😅

**Comment by crisu83** [comment]
@Jalle19 What is what?

File: src/Execution/ValuesResolver.php (hunk 1 above)
Comment (Jalle19) [reply src/Execution/ValuesResolver.php]: There's an optin to ignore the root namespace

File: src/Execution/ValuesResolver.php [hunk 2]
```diff
@@ -558,36 +626,38 @@ protected function printPath(?Path $path)
...
-                    'Argument "%s" of required type "%s" was provided the variable "$%s" which was not provided a runtime value.',
-                    $argumentName,
-                    $argumentType,
-                    $variableName
-                ),
-                [$variableNode]
-            );
-        }
-    }
+//    protected static function coerceValueForVariableNode(
```
Comment (Jalle19) [inline src/Execution/ValuesResolver.php]: Can you remove this instead of commenting it out?

File: src/Execution/ValuesResolver.php [hunk 3]
```diff
@@ -26,13 +27,22 @@
 use Digia\GraphQL\Util\ConversionException;
 use Digia\GraphQL\Util\TypeASTConverter;
 use Digia\GraphQL\Util\ValueASTConverter;
+use Exception;
+use Traversable;
+use function array_key_exists;
```
Comment (Jalle19) [inline src/Execution/ValuesResolver.php]: These still shouldn't be here

File: tests/Functional/Schema/DefinitionPrinterTest.php [hunk 4]
```diff
@@ -33,128 +33,127 @@ public function testPrintStringFields(string $source): void
...
-            ],
-            // String field with Int argument with default value
-            [
-                dedent('
-                type Query {
-                  singleField(argOne: Int = 2): String
-                }
-                '),
-            ],
+//            // String field
```
Comment (Jalle19) [inline tests/Functional/Schema/DefinitionPrinterTest.php]: Why are these commented out?

File: src/Execution/ValuesResolver.php (hunk 1 above)
Comment (crisu83) [reply src/Execution/ValuesResolver.php]: I fixed this and updated the project settings.

File: src/Execution/ValuesResolver.php (hunk 2 above)
Comment (crisu83) [reply src/Execution/ValuesResolver.php]: Sure. I apparently forgot to remove the unused method.

File: src/Execution/ValuesResolver.php (hunk 3 above)
Comment (crisu83) [reply src/Execution/ValuesResolver.php]: This has been fixed.

File: tests/Functional/Schema/DefinitionPrinterTest.php (hunk 4 above)
Comment (crisu83) [reply tests/Functional/Schema/DefinitionPrinterTest.php]: I fixed all the remaining test cases for the definition printer.

**Pull request merged**